kyo-core/shared/src/main/scala/kyo/retries.scala

```diff
@@ -28,18 +28,18 @@ object Retries:
 
     def apply[T: Flat, S](builder: Policy => Policy)(v: => T < S): T < (Fibers & S) =
         val b = builder(Policy.default)
-        def loop(attempt: Int): T < (Fibers & S) =
-            IOs.attempt[T, S](v).map {
+        Loops.indexed { attempt =>
+            IOs.attempt(v).map {
                 case Failure(ex) =>
                     if attempt < b.limit then
                         Fibers.sleep(b.backoff(attempt)).andThen {
-                            loop(attempt + 1)
+                            Loops.continue
                         }
                     else
                         IOs.fail(ex)
-                case Success(value) =>
-                    value
+                case Success(value: T) =>
+                    Loops.done(value)
             }
-        loop(0)
+        }
```

Comment:
Much improved!

Reply:
I’ve been a bit surprised by how well the code reads with a more explicit API like Loops. I’m almost preferring it to regular recursive methods. This might be a good direction to evolve the library: more explicit APIs that don’t incur into runtime cost like in Result, which seems helpful especially to newcomers.